homeassistant/helpers/template.py

```diff
@@ -1529,6 +1530,16 @@ def arc_tangent2(*args, default=_SENTINEL):
         return default
 
 
+def version(value, default=_SENTINEL):
+    """Filter and function to get version object of the value."""
+    try:
+        return AwesomeVersion(value)
+    except AwesomeVersionException:
+        if default is _SENTINEL:
+            raise_no_default("version", value)
+        return default
```

Comment:
The default is interesting in this case.

The question is: Should it be expected that `version()` always returns an `AwesomeVersion` object?

```yaml
{% version(states("sensor.version"), "1.0.0").beta %}
```

☝️ the above would fail in case `sensor.version` isn't a valid version/not set, mainly because the returned default, in that case, is a string (and not an AwesomeVersion object).

Reply:
Hmm...
I do not think default make sense here at all actually

> The question is: Should it be expected that version() always returns an AwesomeVersion object?

I would say yes

Reply:
> I do not think default make sense here at all actually

The default could be useful, but could as well be omitted. There is no much use to do version checks when there is no version at all, that I agree on.